Backpack/src/main/java/net/skyscanner/backpack/button/internal/BpkButtonLayout.kt

```diff
@@ -1,138 +0,0 @@
-
-// responsible for correct compound icon positioning
-abstract class BpkButtonLayout internal constructor(
-  context: Context,
-  attrs: AttributeSet?,
-  defStyleAttr: Int
-) : AppCompatButton(context, attrs, defStyleAttr) {
-
-  private var minPaddingStart = resources.getDimensionPixelSize(R.dimen.bpkSpacingBase) -
-    resources.getDimensionPixelSize(R.dimen.bpkSpacingSm)
-
-  private var minPaddingEnd = minPaddingStart
-  private var maxIconSpacing = resources.getDimensionPixelSize(R.dimen.bpkSpacingSm)
-
-  init {
-    maxLines = 1
-    gravity = Gravity.CENTER
-    ellipsize = TextUtils.TruncateAt.END
-    super.setCompoundDrawablePadding(maxIconSpacing)
-    super.setPaddingRelative(minPaddingStart, paddingTop, minPaddingEnd, paddingBottom)
-  }
-
-  override fun setCompoundDrawablesWithIntrinsicBounds(
-    @DrawableRes left: Int,
-    @DrawableRes top: Int,
-    @DrawableRes right: Int,
-    @DrawableRes bottom: Int
-  ) {
-    super.setCompoundDrawablesWithIntrinsicBounds(left, top, right, bottom)
-    layoutComponents()
-  }
-
-  override fun setCompoundDrawablesWithIntrinsicBounds(
-    left: Drawable?,
-    top: Drawable?,
-    right: Drawable?,
-    bottom: Drawable?
-  ) {
-    super.setCompoundDrawablesWithIntrinsicBounds(left, top, right, bottom)
-    layoutComponents()
-  }
-
-  override fun setCompoundDrawables(left: Drawable?, top: Drawable?, right: Drawable?, bottom: Drawable?) {
-    super.setCompoundDrawables(left, top, right, bottom)
-    layoutComponents()
-  }
-
-  override fun setCompoundDrawablesRelative(start: Drawable?, top: Drawable?, end: Drawable?, bottom: Drawable?) {
-    super.setCompoundDrawablesRelative(start, top, end, bottom)
-    layoutComponents()
-  }
-
-  override fun setText(text: CharSequence, type: BufferType) {
-    super.setText(text, type)
-    layoutComponents()
-  }
-
-  override fun onSizeChanged(w: Int, h: Int, oldw: Int, oldh: Int) {
-    super.onSizeChanged(w, h, oldw, oldh)
-    layoutComponents()
-  }
-
-  override fun setPadding(left: Int, top: Int, right: Int, bottom: Int) {
-    super.setPadding(left, top, right, bottom)
-    minPaddingStart = left
-    minPaddingEnd = right
-    layoutComponents()
-  }
-
-  override fun setPaddingRelative(start: Int, top: Int, end: Int, bottom: Int) {
-    super.setPaddingRelative(start, top, end, bottom)
-    minPaddingStart = start
-    minPaddingEnd = end
-    layoutComponents()
-  }
-
-  override fun setCompoundDrawablePadding(pad: Int) {
-    maxIconSpacing = pad
-    layoutComponents()
-  }
-
-  private fun layoutComponents() {
-    var width = width
-    if (width == 0) {
-      width = measuredWidth
-    }
-    if (width == 0) {
-      return
-    }
-
-    val icon = compoundDrawables[0] ?: compoundDrawables[2]
-    val iconWidth = icon?.intrinsicWidth ?: 0
-    val iconSpacing = if (icon != null && text.isNotEmpty()) maxIconSpacing else 0
-
-    super.setCompoundDrawablePadding(iconSpacing)
-
-    if (icon != null) {
```

Comment:
we used to manually update the button position to align the icon with the text, rather than the end/start of the view. this is now supported by `MaterialButton` so in the next commit i'll change to the materialbutton which also removes a lot of the logic in general